### PR TITLE
"String translated in the past" warning enhancement

### PR DESCRIPTION
**Pull request by 55abhilash**
The string "This string has been translated in the past" changed to "This string has been translated in the past. Last translation was %s".

### CODE REVIEW

**Comment by 55abhilash** [comment]
For issue: #4084

File: weblate/checks/consistency.py [hunk 1]
```diff
@@ -95,9 +95,14 @@ class TranslatedCheck(TargetCheck):
 
     check_id = "translated"
     name = _("Has been translated")
-    description = _("This string has been translated in the past")
+    description = ""
```
Comment (nijel) [inline weblate/checks/consistency.py:98]: Please keep the description here as well, it is used for example when listing all the checks (/checks/ URL).

File: weblate/checks/consistency.py [hunk 2]
```diff
@@ -95,9 +95,14 @@ class TranslatedCheck(TargetCheck):
...
     ignore_untranslated = False
 
+    def get_description(self, check=None):
+        unit = check.unit
+        self.check_target_unit(unit.source, unit.target, unit)
+        return self.description
```
Comment (nijel) [inline weblate/checks/consistency.py:104]: Beware that the in database check state might be outdated, so the check_target_unit might return False when get_description is called. In this case you return random description of previously triggered check.

File: weblate/checks/consistency.py [hunk 3]
```diff
@@ -111,6 +116,13 @@ def check_target_unit(self, sources, targets, unit):
 
         for action in changes.values_list("action", flat=True):
             if action in Change.ACTIONS_CONTENT:
+                self.description = (
+                    _(
+                        'This string has been translated in the past. \
+                                Last Translation was "%s"'
+                    )
+                    % changes.latest("timestamp").target
```
Comment (nijel) [inline weblate/checks/consistency.py:124]: This hits again the database making the check twice slower than necessary. Better approach would be to include target in values_list above and use the value here directly without additional SQL query.

Also changing description class attribute doesn't sound right. Better approach might be to return the target value here and generate the verbose description in the get_description method.

File: weblate/locale/django.pot [hunk 4]
```diff
@@ -2098,7 +2098,7 @@ msgid "Has been translated"
 msgstr ""
 
 #: weblate/checks/consistency.py:98
-msgid "This string has been translated in the past"
+msgid "This string has been translated in the past. Last Translation was \"%s\""
```
Comment (nijel) [inline weblate/locale/django.pot:2101]: This is not needed, the pot/po files are updated time to time by a dedicated script.

File: weblate/checks/consistency.py (hunk 2 above)
Comment (55abhilash) [reply weblate/checks/consistency.py:104]: What is the meaning of "check state might be outdated"

Comment (nijel) [reply weblate/checks/consistency.py:111]: The get_description is called on the translate page when the check object is stored in the database. Meanwhile the string can be updated and check update in progress in the background. So the get_description should work without problems when the check is no longer failing.

File: weblate/checks/consistency.py [hunk 5]
```diff
@@ -98,6 +98,18 @@ class TranslatedCheck(TargetCheck):
     description = _("This string has been translated in the past")
     ignore_untranslated = False
 
+    def get_description(self, check=None):
+        unit = check.unit
+        target = self.check_target_unit(unit.source, unit.target, unit)
+        self.description = (
```
Comment (nijel) [inline weblate/checks/consistency.py:104]: Why updating description attribute? I see no reason for that (and it will break things).

Comment (55abhilash) [reply weblate/checks/consistency.py]: Sorry, missed it before. Changed now.

File: weblate/checks/consistency.py [hunk 6]
```diff
@@ -101,14 +101,14 @@ class TranslatedCheck(TargetCheck):
...
-        self.description = (
+        description = (
             _(
                 'This string has been translated in the past. \
                 Last Translation was "%s"'
             )
             % target
         )
-        return self.description
+        return description
```
Comment (nijel) [inline weblate/checks/consistency.py:111]: ...and now the variable is useless :-). Please use return directly.

File: weblate/checks/consistency.py [hunk 7]
```diff
@@ -98,6 +98,17 @@ class TranslatedCheck(TargetCheck):
     description = _("This string has been translated in the past")
     ignore_untranslated = False
 
+    def get_description(self, check=None):
+        unit = check.unit
+        target = self.check_target_unit(unit.source, unit.target, unit)
+        return (
+            _(
+                'This string has been translated in the past. \
+                Last Translation was "%s"'
```
Comment (nijel) [inline weblate/checks/consistency.py:107]:
```suggestion
                'This string has been translated in the past, last translation was "%s"'
```

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Comment by nijel** [comment]
I've made some futher improvements in 5d7aaaf109e706fea3deecce370e3fd5bf3fb673

**Pull request merged**